curl -L mode; move generic code out of uihandler; more tests; etc...

ldemailly opened the pull request:
- moved generic code from ui to fhttp/
- added `-L` and `FollowRedirects` options (meant for `fortio curl -L url > file`). -L is shortcut for both follow redirect and using the stdclient (decoding the data part, vs raw output from fast client). So `fortio curl -L ...` can be used for downloads.
- `FetchURL` and `Fetch` helper functions wrapping 1 time use client to help write 1 liner readiness
- more tests
- added go version in `fortio version` output (`version.Long()`)
- changed `-httpreqtimeout` flag to just `-timeout`
- expose `fhttp.RedirectToHTTPS`, `NewHTMLEscapeWriter`, `SetupPPROF`, `FetcherHandler` proxy, `LogRequest`, `OnBehalfOf` to the public fhttp library api.
- solved README.md wrapping of fortio command help text using new `release/updateFlags.sh`

ldemailly committed:
[wip, need to add tests] move generic code out of uihandler

ldemailly committed:
Use LogRequest in Echo and Debug

ldemailly committed:
Merge branch 'master' into move_code_add_tests

ldemailly committed:
Adding -L flag (meant for curl mode) to follow redirects and use the std client

And for go side added corresponding
fhttp.FetchURL(url)
And
fhttp.Fetch(httpOPtions)
Returning http status and data, for used as downloaders and/or to check
liveness/initial checks as a better interface to std go fetcher

Ps: tests still upcoming

ldemailly commented:
@mattdelco the `fortio curl -L ...` is for you (and the fhttp.FetchURL() for some of our liveness/readiness tests)

ldemailly committed:
Adding test for pprof and Fetch*

ldemailly committed:
Move test in correct section

mattdelco commented:
Thanks, it looks alright to me (though I don't consider myself very knowledgeable about fortio or golang).  I have been a bit apprehensive about relying on stdout (vs an explicit file output parameter), but so far things seem okay (almost everything else is using stderr, either directly or indirectly via the logging library).

ldemailly commented:
any unix-y program worth its salt will not mix stdout and stderr :-) if it produces something to be saved it will be on stdout while logging goes to stderr

ldemailly committed:
Move the fetcher/proxy to fhttp/ too

Tests upcoming

TODO: check perf / self qps number

ldemailly committed:
Merge branch 'master' into move_code_add_tests

ldemailly committed:
Add the go version to the full version

ldemailly committed:
Change timeout flag name to just -timeout

ldemailly committed:
Merge branch 'master' into move_code_add_tests

# Conflicts:
#	fortio_main.go

ldemailly committed:
New tests for RedirectToHTTPS , HTMLEscapeWriter, OnBehalfOf

ldemailly committed:
Circleci not ipv6

ldemailly committed:
Error instead of fatal and interfacer change

Linter was wrong (as you can’t change from http -> writer with the cast
to flush) but not going to fight it, wrote a better implementation and
test instead

ldemailly committed:
Cover 1 more line (!)

ldemailly committed:
Fix readme formatting/wrapping once for all

Added release/updateFlags.sh
To do the 80 cols wrapping with some sed + fmt magic

ldemailly committed:
update readme for 0.8.0
